src/service/alerts/alert.rs

```diff
@@ -490,6 +490,24 @@ pub async fn list(
     }
 }
 
+pub async fn delete_by_id<C: ConnectionTrait>(
+    conn: &C,
+    org_id: &str,
+    alert_id: Ksuid,
+) -> Result<(), AlertError> {
+    let Some(alert) = db::alerts::alert::get_by_id(conn, org_id, alert_id).await? else {
+        return Ok(());
```

Comment:
Should this return NotFound error instead, or at least according to the route `#[delete("v2/{org_id}/alerts/{alert_id}")]`?

Reply:
Yeah, I think we should either (a) return an error here or (b) simply not do the check at all.

My preference, personally, is for (b) since I don't think there's really much benefit to the client from returning an error. It just costs us an extra DB call and it makes the operation non-idempotent. 

However if we do actually want to return an error here, we should probably also wrap the "get" and "delete" operations inside a transaction since otherwise there is nothing guaranteeing that the record still exists when the "delete" operation is called after the "get" operation.